doc/user-guide/muc_light.md

```diff
@@ -0,0 +1,1821 @@
+    <body></body>
+</message>
+```
+
+```xml
+<message from='coven@muclight.shakespeare.lit'
+         to='hag77@shakespeare.lit'
+         type='groupchat'
+         id='memberchange'>
+    <x xmlns='urn:xmpp:muclight:0#affiliations'>
+        <prev-version>njiokm</prev-version>
+        <version>qwerty</version>
+        <user affiliation='member'>crone1@shakespeare.lit</user>
+        <user affiliation='member'>hag66@shakespeare.lit</user>
+        <user affiliation='owner'>hag77@shakespeare.lit</user>
+        <user affiliation='none'>hag88@shakespeare.lit</user>
+    </x>
+    <body></body>
+</message>
+```
+
+```xml
+<message from='coven@muclight.shakespeare.lit'
+         to='hag88@shakespeare.lit'
+         type='groupchat'
+         id='memberchange'>
+    <x xmlns='urn:xmpp:muclight:0#affiliations'>
+        <user affiliation='none'>hag88@shakespeare.lit</user>
+    </x>
+    <body></body>
+</message>
+```
+
+```xml
+<iq to='crone1@shakespeare.lit/desktop'
+    id='member1'
+    from='coven@muclight.shakespeare.lit'
+    type='result' />
+```
+
+# 6. Interactions with RFCs and other XEPs
+
+## 6.1 User rosters
+
+The service MAY add user's rooms to its roster. It allows the client to skip separate Disco request to the service. Roster items with rooms MUST belong to the group "urn:xmpp:muclight:0" (MUC Light namespace) and include <version> element. Their subscription type MUST be 'to'.
+
+**Entity requests the roster and receives reply that includes room item**
+
+```xml
+<iq type='get' id='roster1' to='shakespeare.lit'>
+    <query xmlns='jabber:iq:roster'/>
+</iq>
+```
+
+```xml
+<iq id='roster1' to='hag66@shakespeare.lit/tablet' type='result'>
+    <query xmlns='jabber:iq:roster' ver='ver7'>
+        <item jid='hag77@shakespeare.lit' subscription='both'/>
+        <item jid='hag88@shakespeare.lit' subscription='both'/>
+        <item jid='coven@muclight.shakespeare.lit' name='The Coven' subscription='to'>
+            <group>urn:xmpp:muclight:0</group>
+            <version>1234345</version>
+        </item>
+    </query>
+</iq>
+```
+
+## 6.2 XEP-0313 Message Archive Management
+
+This section defines the rules for archiving MUC Light events and messages. Stanzas described in the subsections below MUST be archived by the server. The stanzas not included here MUST NOT be archived.
+
+The <message/> element inside <forwarded> MUST include "from" attribute and MUST NOT include "to" attribute. "id" SHOULD be archived as well.
+
+In case of regular groupchat messages, "from" attribute MUST consist of a room full JID with sender bare JID in the resource part. As for room notification, e.g. create event, "from" MUST be equal to room bare JID.
+
+** protocol. Archive can be fetched only from a specific room, the client MUST NOT query MUC Light service directly.**
+
+### 6.2.1 Groupchat message from occupant
+
+Message from a user MUST be archived with all child elements.
+
+**Occupant queries MAM and receives regular groupchat message**
```

Comment:
I'd leave it here and work on fixing the implementation. I think this is a must-have feature.